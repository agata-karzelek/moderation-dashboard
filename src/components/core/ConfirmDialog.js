import React from "react";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import { useStyles } from "../../style/confirmDialog";
import { Typography } from "@material-ui/core";

const ConfirmDialog = (props) => {
  const { title, children, open, setOpen, onConfirm } = props;
  const classes = useStyles();

  return (
    <Dialog
      id="dialog"
      open={open}
      onClose={() => setOpen(false)}
      aria-labelledby="confirm-dialog"
      fullWidth
      maxWidth="xs"
      className={classes.confirmDialog}
    >
      <DialogTitle id="confirm-dialog">{title}</DialogTitle>
      <DialogContent id="content">{children}</DialogContent>
      <DialogActions>
        <Button
          id="confirm"
          onClick={() => {
            setOpen(false);
            onConfirm();
          }}
          className={classes.confirmButton}
        >
          <Typography>{props.actionMessage}</Typography>
        </Button>

        <Button
          id="cancel"
          variant="contained"
          onClick={() => setOpen(false)}
          className={classes.cancelButton}
        >
          Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
};
export default ConfirmDialog;
