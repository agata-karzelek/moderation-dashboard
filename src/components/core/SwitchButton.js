import Switch from "@material-ui/core/Switch";
import SwitchButtonStyle from "../../style/switchButton";

const SwitchButton = SwitchButtonStyle(({ classes, ...props }) => {
  return (
    <Switch
      focusVisibleClassName={classes.focusVisible}
      disableRipple
      classes={{
        root: classes.root,
        switchBase: classes.switchBase,
        thumb: classes.thumb,
        track: classes.track,
        checked: classes.checked,
      }}
      {...props}
    />
  );
});

export default SwitchButton;
