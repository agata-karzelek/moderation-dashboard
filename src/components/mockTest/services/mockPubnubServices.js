import { mockAccounts } from "../mockAccounts";
import { mockPermissions } from "../mockPermissions";
import { mockPubNubApplications } from "../mockPubnubAccounts";

const commonkeyValuePairs = {
  status: 200,
  statusText: "OK",
  headers: {
    "cache-control": "no-cache, private",
    "content-type": "application/json",
    expires: "Thu, 29 Jul 2021 15:37:05 GMT",
    pragma: "private",
  },
  config: {
    url: "/apps?owner_id=572238",
    method: "get",
    headers: {
      Accept: "application/json, text/plain, */*",
      "X-Session-Token": "rAL18qRqj8rhtxkJwzQ2dmRi5UrhH-vFbqUaLknxshs",
    },
    baseURL: "https://admin.pubnub.com/api",
    transformRequest: [null],
    transformResponse: [null],
    timeout: 0,
    xsrfCookieName: "XSRF-TOKEN",
    xsrfHeaderName: "X-XSRF-TOKEN",
    maxContentLength: -1,
    maxBodyLength: -1,
  },
  request: {},
};

export const mockAllAccounts = {
  data: {
    result: {
      accounts: mockAccounts,
      permissions: mockPermissions,
    },
  },
  ...commonkeyValuePairs,
};

export const mockAllApps = {
  data: mockPubNubApplications,
  ...commonkeyValuePairs,
};

export default null;
