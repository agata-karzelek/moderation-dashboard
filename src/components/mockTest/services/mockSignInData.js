import { mockPermissions } from "../mockPermissions";

export const mockSignInData = {
  data: {
    result: {
      created: 1627276598,
      expires: 1627634943,
      modified: 1627548543,
      role: "user",
      status: 1,
      storage: null,
      token: "dGgB0-lLGG5GmjFgtr6t2nBFqgM05Nijq_YzzYdxVUk",
      user: {
        created: 1619760527,
        deleted: 0,
        email: "sourcefusepubnub@gmail.com",
        id: 572287,
        login_attempts: 0,
      },
      user_id: 572287,
      user_roles: {
        is_internal_admin: false,
        is_internal_operator: false,
        is_internal_viewer: false,
        roles: mockPermissions,
      },
    },
  },
  status: 200,
  statusText: "OK",
};

export default null;
