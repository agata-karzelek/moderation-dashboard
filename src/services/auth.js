import axios from "../utils/axios";

export async function signIn(credentials) {
  const signInResponse = await axios.post("/me", credentials);
  if (signInResponse.status === 200) {
    return signInResponse.data;
  }
  throw new Error(signInResponse.data.error || "Something went wrong");
}
