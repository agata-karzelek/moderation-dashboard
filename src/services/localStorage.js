import cookie from "js-cookie";

export const setCookie = (key, value) => {
  cookie.set(key, value, {
    expires: 1,
  });
};

export const removeCookie = (key) => {
  cookie.remove(key, {
    expires: 1,
  });
};

export const getCookie = (key) => {
  return cookie.get(key);
};

export const setLocalStorage = (key, value) => {
  localStorage.setItem(key, JSON.stringify(value));
};

export const removeLocalStorage = (key) => {
  localStorage.removeItem(key);
};

export const authenticate = (data, next) => {
  setCookie("token", data.token);
  setLocalStorage("user", data.user);
  next();
};

export const signout = (next) => {
  localStorage.clear();
  next();
};

export const isAuth = () => {
  const cookieChecked = getCookie("token");
  if (cookieChecked) {
    if (localStorage.getItem("user")) {
      return JSON.parse(localStorage.getItem("user"));
    } else {
      return false;
    }
  }
};
