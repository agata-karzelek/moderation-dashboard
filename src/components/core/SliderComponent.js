import React from "react";
import { useStyles, SlideBar } from "../../style/slideBar";
import { Slider } from "@material-ui/core";

const PrettoSlider = SlideBar(Slider);

export default function SliderComponent({ value, onChange, name }) {
  const classes = useStyles();

  const marks = [
    {
      value: 0,
      label: name === "tisane" ? "None" : 0,
    },
    {
      value: 0.25,
      label: name === "tisane" ? "Extreme" : 0.25,
    },
    {
      value: 0.5,
      label: name === "tisane" ? "High" : 0.5,
    },
    {
      value: 0.75,
      label: name === "tisane" ? "Medium" : 0.75,
    },
    {
      value: 1,
      label: name === "tisane" ? "Low" : 1,
    },
  ];

  function valuetext(values) {
    return `${values}`;
  }

  return (
    <div className={classes.root}>
      <PrettoSlider
        value={value}
        min={0}
        getAriaValueText={valuetext}
        aria-labelledby="discrete-slider"
        step={0.25}
        marks={marks}
        max={1}
        onChange={onChange}
      />
    </div>
  );
}
