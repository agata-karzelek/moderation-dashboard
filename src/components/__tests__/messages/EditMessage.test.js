import React from "react";
import { shallow, mount } from "enzyme";
import EditMessage from "../../messages/EditMessage";
import { IconButton, TextField } from "@material-ui/core";
import { addEditMessageAction, deleteMessageAction } from "../../../services/pubnub";

const mockUpdateFn = jest.fn();
jest.mock("../../../services/pubnub", () => ({
  addEditMessageAction: jest.fn(),
  deleteMessageAction: jest.fn(),
}));

describe("Test Case For Chat", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <EditMessage
        channel="pro.one"
        message={{
          text: "abc",
          timetoken: 123,
        }}
        updated={mockUpdateFn}
        messagesLength={30}
        pubnub=""
      />
    );
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("check handleInputChange", async () => {
    expect(wrapper.find("#message").getElement().props.value).toBe("");
    await wrapper.find("#message").simulate("change", { target: { value: "newText" } });
    expect(wrapper.find(TextField).getElement().props.value).toBe("newText");

    deleteMessageAction.mockImplementationOnce(() => [{}]);
    addEditMessageAction.mockImplementationOnce(() => [{}]);
    wrapper
      .find("#message")
      .getElement()
      .props.InputProps.endAdornment.props.children.props.onClick();
    expect(addEditMessageAction).toHaveBeenCalledTimes(1);
  });

  test("check functions", async () => {
    wrapper = mount(
      <EditMessage
        channel="pro.one"
        message={{
          text: "abc",
          timetoken: 123,
        }}
        updated={mockUpdateFn}
        messagesLength={30}
        pubnub=""
      />
    );

    expect(mockUpdateFn).toHaveBeenCalledTimes(0);
    wrapper.find(IconButton).at(0).simulate("click");
    expect(mockUpdateFn).toHaveBeenCalledTimes(1);
  });
});
