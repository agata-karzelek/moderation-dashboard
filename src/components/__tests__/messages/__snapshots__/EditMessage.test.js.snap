// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For Chat Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "endAdornment": <WithStyles(ForwardRef(InputAdornment))
          position="end"
        >
          <WithStyles(ForwardRef(Button))
            onClick={[Function]}
          >
            <img
              alt="edit"
              src="/images/send-button.svg"
            />
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(InputAdornment))>,
      }
    }
    autoComplete="off"
    disabled={true}
    fullWidth={true}
    id="message"
    name="message"
    onChange={[Function]}
    placeholder="Enter your message here"
    value=""
    variant="outlined"
  />
</Fragment>
`;
